### PR TITLE
#30 course menagament fix get all courses by search term

### PR DESCRIPTION
**Pull request by BojanMirkovic**
#  Fix get all courses by search term

## Description

I have changed and modified GetAllCorsesBySearchCriteria 

## Changes Made
Changes I made: CourseControler method name, GetAllCoursesBySearchCriteriaQueryHandler, ICorseRepository and CourseRepository.
All tests are updated.

## Checklist

- [x] Code adheres to the coding conventions
- [x] Unit tests are added for new code
- [x] Existing unit tests are updated if needed
- [x] Code follows the principles of Clean Architecture
- [x] Code formatting complies with the `.editorconfig` file
- [x] Pull request template is filled out

### CODE REVIEW

**Review by ElliotDahlin — APPROVED** [review]
everything looks good to me i see that you added getall incase somebody searches for nothing and that you removed the handling code from repo into the handler, i did not see you moving that model/Dto in repo ? could you maybe make a Dto based on properties from both user and course and then configure your method in repo to use usermodel and coursemodel where needed??

**Comment by BojanMirkovic** [comment]
> everything looks good to me i see that you added getall incase somebody searches for nothing and that you removed the handling code from repo into the handler, i did not see you moving that model/Dto in repo ? could you maybe make a Dto based on properties from both user and course and then configure your method in repo to use usermodel and coursemodel where needed??

File: Application-Layer/Queries/CourseQueries/GetAllCoursesBySearchCriteria/GetAllCoursesBySearchCriteriaQueryHandler.cs [hunk 1]
```diff
@@ -12,23 +11,43 @@ public GetAllCoursesBySearchCriteriaQueryHandler(ICourseRepository courseReposit
...
             _courseRepository = courseRepository;
         }
-
         public async Task<List<CourseModel>> Handle(GetAllCoursesBySearchCriteriaQuery request, CancellationToken cancellationToken)
         {
-            if (request.SearchCriteriaInfo == null)
+            try
             {
-                throw new ArgumentException("Course with searched criteria, was not found!");
+                // SearchCriteria is a property of the request object
```
Comment (Nemanja1208) [inline Application-Layer/Queries/CourseQueries/GetAllCoursesBySearchCriteria/GetAllCoursesBySearchCriteriaQueryHandler.cs:18]: Perfekt but remove comments

**Review by Nemanja1208 — APPROVED** [review]
Remove comments and this is Guci

**Pull request merged**